lib/drivers/medtronic600Driver.js

```diff
@@ -0,0 +1,1221 @@
+    }
+    /* eslint-enable lodash/prefer-lodash-method */
+
+    // Now that we have written the message, calculate the CRC
+    const checksum = MinimedMessage.oneByteChecksum(payloadBuffer);
+    payloadBuffer.writeUInt8(checksum, 32);
+    return payloadBuffer;
+  }
+}
+
+class TelDMessage extends MinimedMessage {
+  static get COMMAND_TYPE() {
+    return {
+      INITIALIZE: 0x01,
+      SCAN_NETWORK: 0x02,
+      JOIN_NETWORK: 0x03,
+      LEAVE_NETWORK: 0x04,
+      TRANSMIT_PACKET: 0x05,
+      READ_DATA: 0x06,
+      READ_STATUS: 0x06,
+      READ_NETWORK_STATUS: 0x06,
+      SET_SECURITY_MODE: 0x0c,
+      READ_STATISTICS: 0x0d,
+      SET_RF_MODE: 0x0e,
+      CLEAR_STATUS: 0x10,
+      SET_LINK_KEY: 0x14,
+      PACKET_RESPONSE_HEADER: 0x55, // 'U'
+    };
+  }
+
+  static ccittChecksum(buffer, length) {
+    return crcCalculator.calcCRC_A(buffer, length);
+  }
+}
+
+class TelDResponse extends TelDMessage {
+  // TODO check response sizes, checksums, etc
+}
+
+class TelDRequest extends TelDMessage {
+  constructor(commandType, pumpSession, payload) {
+    if (new.target === TelDRequest) {
+      throw new TypeError(`Cannot construct ${new.target.name} instances directly`);
+    }
+
+    super(TelDRequest.buildPayload(commandType, pumpSession, payload));
+    // TODO - do we want to do this everywhere??
+    this.pumpSession = pumpSession;
+  }
+
+  static get ENVELOPE_SIZE() {
+    return 2;
+  }
+
+  static get CHECKSUM_SIZE() {
+    return 2;
+  }
+
+  static buildPayload(commandType, pumpSession, payload) {
+    const payloadLength = (payload === null) ? 0 : payload.length;
+    const payloadBuffer = Buffer.alloc(
+      TelDRequest.ENVELOPE_SIZE + payloadLength + TelDRequest.CHECKSUM_SIZE);
+
+    payloadBuffer.writeUInt8(commandType, 0);
+    payloadBuffer.writeUInt8(TelDRequest.ENVELOPE_SIZE + payloadLength, 1);
+    if (payloadLength > 0) {
+      payloadBuffer.write(Buffer.from(payload).toString('binary'), 2, 'binary');
+    }
+
+    // Now that we have written the message, calculate the CRC
+    const messageSize = payloadBuffer.length - 2;
+    const checksum = TelDMessage.ccittChecksum(payloadBuffer, messageSize);
+    payloadBuffer.writeUInt16LE(checksum, messageSize);
+
+    return MinimedRequest.buildPayload(MinimedRequest.COMMAND_TYPE.SEND_MESSAGE,
+      pumpSession, payloadBuffer);
+  }
+
+  // Override send(), because we do an 'optional double read' after sending the request
+  send(hidDevice, readTimeout = BCNLMessage.READ_TIMEOUT_MS, sendDelay = 0, get80response = true) {
+    // eslint-disable-next-line no-unused-vars
+    return new Promise((resolve, reject) => {
+      // Timeout might be zero, but the callback will fire anyway
+      this.sendMessage(hidDevice)
+        .then(() => BCNLMessage.wait(sendDelay))
+        .then(() => this.readMessage(hidDevice, readTimeout))
+        .then((response) => {
+          // TODO - throw here if the 0x81 doesn't check out.
```

Comment:
Yes, TODO's will be dealt with as part of the "production readiness" Tello card :)